pkg/gsm-secrets/secrets.go

```diff
@@ -8,9 +8,10 @@ import (
 	"github.com/openshift/ci-tools/pkg/group"
 )
 
-// ValidateSecretName validates if a secret name matches the allowed pattern
+// ValidateSecretName validates if a secret name matches the allowed pattern.
+// The secret name must be between 1 and 255 characters long (GCP limit).
 func ValidateSecretName(secretName string) bool {
-	return regexp.MustCompile(SecretNameRegex).MatchString(secretName)
+	return regexp.MustCompile(SecretNameRegex).MatchString(secretName) && len(secretName) <= 255
```

Comment:
I would recommend replacing `255` with a constant name like `GCP_MAX_NAME_LENGTH` or something to that effect.